core/contracts/types/entitlement_data.go

```diff
@@ -155,9 +143,29 @@ func MarshalEntitlement(
 			EntitlementType:   rawEntitlement.EntitlementType,
 			RuleEntitlementV2: &ruleData,
 		}, nil
-	} else {
-		return Entitlement{}, fmt.Errorf("invalid entitlement type '%s'", rawEntitlement.EntitlementType)
 	}
+
+	if rawEntitlement.EntitlementType == ModuleTypeUserEntitlement {
```

Comment:
was there a reason to kill the else/if continuity here? I realize it's technically the same code, but it's harder to maintain and harder to reason about flow if there are discrete blocks of code as apposed to one control block. 

If you don't have a good reason i suggest adding it back.

Reply:
I think it just helped me fix the indentation from all of my copy pasta and I left it that way thinking it was good either way... happy to convert it back!